screen19/minimum_exposure.py

```diff
@@ -304,13 +291,38 @@ def suggest_minimum_exposure(expts, refls, params):
     refls.del_selected(refls["id"] == -1)
     # Ignore all spots flagged as overloaded
     refls.del_selected(refls.get_flags(refls.flags.overloaded).iselection())
-    # The Wilson plot fit implicitly involves taking a logarithm of
-    # intensities, so eliminate values that are going to cause problems
-    try:
-        # Work from profile-fitted intensities where possible
-        refls = refls.select(refls["intensity.prf.value"] > 0)
-    except RuntimeError:
-        refls = refls.select(refls["intensity.sum.value"] > 0)
+
+    # Work from profile-fitted intensities where possible but if the number of
+    # profile-fitted intensities is less than 75% of the number of summed
+    # intensities, use summed intensities instead.
+    sel_prf = refls.get_flags(refls.flags.integrated_prf).iselection()
+    sel_sum = refls.get_flags(refls.flags.integrated_sum).iselection()
+    if sel_prf.size() < 0.75 * sel_sum.size():
```

Comment:
Got some reasoning for `0.75` here?

Reply:
In all honesty I would wonder how much difference profile vs. summation would make here anyways

Reply:
> Got some reasoning for `0.75` here?

I'm glad you mention it.  No.  Basically, I don't know enough about how `dials.integrate` sets flags to know its typical success rates for each of the two integration routines.  Basically, if there are a significant number of reflections that have been profile-fit-integrated successfully, I want to use profile fitting.  If a substantial number have failed, I want to fall back on summation integration.  I have no better idea of how to achieve this (except perhaps by calculating an R-factor for each of the summation- and profile-fit-integration strategies and selecting on that, or something like it?).  `0.75` is a brute force and stupidity approach.

> In all honesty I would wonder how much difference profile vs. summation would make here anyways

Given that we know that profile-fit-integration will work better for cases with diffraction artefacts such as diamond anvil reflections, I think we do want to try to use it, unless it's not sensible, as determined by some heuristic (stupid or otherwise).

Reply:
I can't see any clever way to do this.  Of course R-values won't work — in the case of a set of reflections that are unique under symmetry, such as one might expect from a single image, the R-meas and R-merge of both the summation integrated and profile-fitted integrated intensities will obviously be zero.  I'm going to leave the ugly heuristic of 75%.  If it causes problems in future, we can deal with it then.